Adding Makefiles to build libraries on Linux/macOS

bucanero opened the pull request:
This PR updated the lib Makefiles so they work with Open Orbis SDK on Linux and macOS.

Notes: 

- `build_rules.mk` must be copied to `$OO_PS4_TOOLCHAIN` folder. For example, `/opt/OpenOrbis-PS4-Toolchain/build_rules.mk`. 
- I also included a minor fix to `zlib` ( madler/zlib#183)

If required, I can submit a PR to the main repo to add `build_rules.mk` there.

bucanero committed:
Add makefiles

bucanero committed:
Fix clang shifting issue

shifting a negative signed value is undefined ("-1L << 16" in inflate.c)
https://github.com/madler/zlib/issues/183

bucanero committed:
Add missing include

include for ioctl(...)

bucanero committed:
Add polarSSL v1.3.9

bucanero committed:
Fix C++ rule

sleirsgoevy commented:
> ... must be copied to ...

What if its installed system-wide and I don't have root?

bucanero commented:
> > ... must be copied to ...
> 
> What if its installed system-wide and I don't have root?

Yes, I see your point.
I think that a `build_rules.mk` should be part of the OpenOrbis repository and be bundled with the installation packages. That way, if you install Open Orbis SDK these generic build rules would be available for any Makefile and any Open Orbis project. I think that would also simplify most OO makefiles a lot, as many steps could be moved to this file.

bucanero committed:
patch for PS Vita

bucanero committed:
Update Makefile

llvm 12

bucanero committed:
Update Makefile

llvm 12

bucanero committed:
add libcurl 7.64.1

bucanero committed:
add polarSSL to cmake

bucanero committed:
add more platforms

bucanero committed:
add libzip 1.9.2

tonytwostep committed:
Update minimum CMake version requirement to 3.10 (#1)

recent-ish requirement, will fail to build on macos without this change (and similar change to SDL-PS4's openorbis.cmake)

* Update minimum CMake version to 3.10

this suppresses the warning as well as the error
